.github/workflows/continuous-integration.yml

```diff
@@ -69,9 +75,58 @@ jobs:
       - name: "Run a static analysis with vimeo/psalm"
         run: "vendor/bin/psalm --show-info=false --stats --output-format=github --threads=4"
 
+  phpunit-smoke-check:
+    name: "PHPUnit with SQLite"
+    runs-on: "ubuntu-20.04"
+
+    strategy:
+      matrix:
+        php-version:
+          - "7.3"
+          - "7.4"
+        deps:
+          - fixed
```

Comment:
My OCD is still unhappy. What's your take on quoting stings?

Reply:
IIRC `no` can be wrongly converting to `false`. Let's not take any chances and quote everything.

Reply:
FWIW, this is what the version that quotes only the necessary pieces looks like:

<details>
  <summary>Click to expand!</summary>
  
```yaml
name: Continuous Integration
on:
  pull_request:
    branches: '**'
  push:
    branches: '**'
  schedule:
  - cron: 42 3 * * *
jobs:
  development-deps:
    name: PHPUnit with SQLite and development dependencies
    runs-on: ubuntu-20.04
    steps:
    - name: Checkout
      uses: actions/checkout@v2
    - name: Install PHP
      uses: shivammathur/setup-php@v2
      with:
        php-version: ${{ matrix.php-version }}
    - name: Cache dependencies installed with composer
      uses: actions/cache@v2
      with:
        key: php-${{ matrix.php-version }}-composer-locked-${{ hashFiles('composer.lock')
          }}
        path: ~/.composer/cache
        restore-keys: php-${{ matrix.php-version }}-composer-locked-
    - name: Lower minimum stability
      run: composer config minimum-stability dev
    - name: Install development dependencies with composer
      run: composer update --no-interaction --no-progress --no-suggest --prefer-dist
    - name: Run PHPUnit
      run: vendor/bin/phpunit -c ci/github/phpunit.sqlite.xml
    strategy:
      matrix:
        php-version:
        - "7.4"
  phpunit-ibm-db2:
    name: PHPUnit with IBM DB2
    needs: phpunit-smoke-check
    runs-on: ubuntu-18.04
    services:
      ibm_db2:
        env:
          DB2INST1_PASSWORD: Doctrine2018
          DBNAME: doctrine
          LICENSE: accept
        image: ibmcom/db2:11.5.0.0
        options: --privileged=true
        ports:
        - 50000:50000
    steps:
    - name: Perform healthcheck from the outside
      run: docker logs -f ${{ job.services.ibm_db2.id }} | sed '/(*) Setup has completed./
        q'
    - name: Create temporary tablespace
      run: docker exec ${{ job.services.ibm_db2.id }} su - db2inst1 -c 'db2 CONNECT
        TO doctrine && db2 CREATE USER TEMPORARY TABLESPACE doctrine_tbsp PAGESIZE
        4 K'
    - name: Checkout
      uses: actions/checkout@v2
      with:
        fetch-depth: 2
    - name: Install PHP
      uses: shivammathur/setup-php@v2
      with:
        coverage: pcov
        ini-values: extension=ibm_db2.so, ibm_db2.instance_name=db2inst1
        php-version: ${{ matrix.php-version }}
    - name: Install ibm_db2 extension
      run: bash ./ci/github/install-db2-ibm_db2.sh ${{ matrix.php-version }}
    - name: Cache dependencies installed with composer
      uses: actions/cache@v2
      with:
        key: php-${{ matrix.php-version }}-composer-locked-${{ hashFiles('composer.lock')
          }}
        path: ~/.composer/cache
        restore-keys: php-${{ matrix.php-version }}-composer-locked-
    - name: Install dependencies with composer
      run: composer install --no-interaction --no-progress --no-suggest
    - name: Run PHPUnit
      run: vendor/bin/phpunit -c ci/github/phpunit.ibm_db2.xml --coverage-clover=coverage.xml
    - name: Upload Code Coverage
      uses: codecov/codecov-action@v1
    strategy:
      matrix:
        php-version:
        - "7.3"
  phpunit-mariadb:
    name: PHPUnit with MariaDB
    needs: phpunit-smoke-check
    runs-on: ubuntu-20.04
    services:
      mariadb:
        env:
          MYSQL_ALLOW_EMPTY_PASSWORD: true
          MYSQL_DATABASE: doctrine_tests
        image: mariadb:${{ matrix.mariadb-version }}
        options: --health-cmd "mysqladmin ping --silent"
        ports:
        - 3306:3306
    steps:
    - name: Checkout
      uses: actions/checkout@v2
      with:
        fetch-depth: 2
    - name: Install PHP
      uses: shivammathur/setup-php@v2
      with:
        coverage: pcov
        extensions: ${{ matrix.driver }}
        php-version: ${{ matrix.php-version }}
    - name: Cache dependencies installed with composer
      uses: actions/cache@v2
      with:
        key: php-${{ matrix.php-version }}-composer-locked-${{ hashFiles('composer.lock')
          }}
        path: ~/.composer/cache
        restore-keys: php-${{ matrix.php-version }}-composer-locked-
    - name: Install dependencies with composer
      run: composer install --no-interaction --no-progress --no-suggest
    - name: Run PHPUnit
      run: vendor/bin/phpunit -c ci/github/phpunit.${{ matrix.driver }}.xml --coverage-clover=coverage.xml
    - name: Upload Code Coverage
      uses: codecov/codecov-action@v1
    strategy:
      matrix:
        driver:
        - mysqli
        - pdo_mysql
        mariadb-version:
        - "10.0"
        - "10.1"
        - "10.2"
        - "10.3"
        php-version:
        - "7.4"
  phpunit-mssql:
    name: PHPUnit with MSSQL
    needs: phpunit-smoke-check
    runs-on: ubuntu-20.04
    services:
      mssql:
        env:
          ACCEPT_EULA: "Y"
          MSSQL_COLLATION: ${{ matrix.collation }}
          SA_PASSWORD: Doctrine2018
        image: microsoft/mssql-server-linux:2017-latest
        options: --health-cmd "echo quit | /opt/mssql-tools/bin/sqlcmd -S 127.0.0.1
          -l 1 -U sa -P Doctrine2018"
        ports:
        - 1433:1433
    steps:
    - name: Checkout
      uses: actions/checkout@v2
      with:
        fetch-depth: 2
    - name: Install PHP
      uses: shivammathur/setup-php@v2
      with:
        coverage: pcov
        extensions: ${{ matrix.driver }}-5.7.0preview
        php-version: ${{ matrix.php-version }}
        tools: pecl
    - name: Cache dependencies installed with composer
      uses: actions/cache@v2
      with:
        key: php-${{ matrix.php-version }}-composer-locked-${{ hashFiles('composer.lock')
          }}
        path: ~/.composer/cache
        restore-keys: php-${{ matrix.php-version }}-composer-locked-
    - name: Install dependencies with composer
      run: composer install --no-interaction --no-progress --no-suggest
    - name: Run PHPUnit
      run: vendor/bin/phpunit -c ci/github/phpunit.${{ matrix.driver }}.xml --coverage-clover=coverage.xml
    - name: Upload Code Coverage
      uses: codecov/codecov-action@v1
    strategy:
      matrix:
        collation:
        - Latin1_General_100_CI_AS
        driver:
        - sqlsrv
        - pdo_sqlsrv
        include:
        - collation: Latin1_General_100_CS_AS
          driver: sqlsrv
          php-version: "7.4"
        - collation: Latin1_General_100_CS_AS
          driver: pdo_sqlsrv
          php-version: "7.4"
        php-version:
        - "7.3"
        - "7.4"
  phpunit-mysql:
    name: PHPUnit with MySQL
    needs: phpunit-smoke-check
    runs-on: ubuntu-20.04
    services:
      mysql:
        image: mysql:${{ matrix.mysql-version }}
        options: --health-cmd "mysqladmin ping --silent" -e MYSQL_ALLOW_EMPTY_PASSWORD=yes
          -e MYSQL_DATABASE=doctrine_tests ${{ matrix.custom-entrypoint }}
        ports:
        - 3306:3306
    steps:
    - name: Checkout
      uses: actions/checkout@v2
      with:
        fetch-depth: 2
    - name: Install PHP
      uses: shivammathur/setup-php@v2
      with:
        coverage: pcov
        extensions: ${{ matrix.driver }}
        php-version: ${{ matrix.php-version }}
    - name: Cache dependencies installed with composer
      uses: actions/cache@v2
      with:
        key: php-${{ matrix.php-version }}-composer-locked-${{ hashFiles('composer.lock')
          }}
        path: ~/.composer/cache
        restore-keys: php-${{ matrix.php-version }}-composer-locked-
    - name: Install dependencies with composer
      run: composer install --no-interaction --no-progress --no-suggest
    - if: ${{ endsWith(matrix.config-file, 'tls') }}
      name: Copy TLS-related files
      run: docker cp "${{ job.services.mysql.id }}:/var/lib/mysql/ca.pem" . && docker
        cp "${{ job.services.mysql.id }}:/var/lib/mysql/client-cert.pem" . && docker
        cp "${{ job.services.mysql.id }}:/var/lib/mysql/client-key.pem" .
    - name: Run PHPUnit
      run: vendor/bin/phpunit -c ci/github/${{ matrix.config-file-prefix }}.${{ matrix.driver
        }}.xml --coverage-clover=coverage.xml
    - name: Upload Code Coverage
      uses: codecov/codecov-action@v1
    strategy:
      matrix:
        config-file-prefix:
        - phpunit
        driver:
        - mysqli
        - pdo_mysql
        include:
        - mysql-version: "5.7"
        - custom-entrypoint: --entrypoint sh mysql:8 -c "exec docker-entrypoint.sh
            mysqld --default-authentication-plugin=mysql_native_password"
          mysql-version: "8.0"
        - config-file-prefix: phpunit.mysql-tls
          driver: mysqli
          mysql-version: "8.0"
          php-version: "7.4"
        mysql-version:
        - "5.7"
        - "8.0"
        php-version:
        - "7.3"
        - "7.4"
  phpunit-oci8:
    name: PHPUnit on OCI8
    needs: phpunit-smoke-check
    runs-on: ubuntu-20.04
    services:
      oracle:
        image: wnameless/oracle-xe-11g-r2
        ports:
        - 1521:1521
    steps:
    - name: Checkout
      uses: actions/checkout@v2
      with:
        fetch-depth: 2
    - name: Install PHP
      uses: shivammathur/setup-php@v2
      with:
        coverage: pcov
        extensions: oci8
        php-version: ${{ matrix.php-version }}
    - name: Cache dependencies installed with composer
      uses: actions/cache@v2
      with:
        key: php-${{ matrix.php-version }}-composer-locked-${{ hashFiles('composer.lock')
          }}
        path: ~/.composer/cache
        restore-keys: php-${{ matrix.php-version }}-composer-locked-
    - name: Install dependencies with composer
      run: composer install --no-interaction --no-progress --no-suggest
    - name: Run PHPUnit
      run: vendor/bin/phpunit -c ci/github/phpunit.oci8.xml --coverage-clover=coverage.xml
    - name: Upload Code Coverage
      uses: codecov/codecov-action@v1
    strategy:
      matrix:
        php-version:
        - "7.4"
  phpunit-pdo-oci:
    name: PHPUnit on PDO_OCI
    needs: phpunit-smoke-check
    runs-on: ubuntu-20.04
    services:
      oracle:
        image: wnameless/oracle-xe-11g-r2
        ports:
        - 1521:1521
    steps:
    - name: Checkout
      uses: actions/checkout@v2
      with:
        fetch-depth: 2
    - name: Install PHP
      uses: shivammathur/setup-php@v2
      with:
        coverage: pcov
        extensions: pdo_oci
        php-version: ${{ matrix.php-version }}
    - name: Cache dependencies installed with composer
      uses: actions/cache@v1
      with:
        key: php-${{ matrix.php-version }}-composer-locked-${{ hashFiles('composer.lock')
          }}
        path: ~/.composer/cache
        restore-keys: php-${{ matrix.php-version }}-composer-locked-
    - name: Install dependencies with composer
      run: composer install --no-interaction --no-progress --no-suggest
    - name: Run PHPUnit
      run: vendor/bin/phpunit -c ci/github/phpunit.pdo-oci.xml --coverage-clover=coverage.xml
    - name: Upload Code Coverage
      uses: codecov/codecov-action@v1
    strategy:
      matrix:
        php-version:
        - "7.4"
  phpunit-postgres:
    name: PHPUnit with PostgreSQL
    needs: phpunit-smoke-check
    runs-on: ubuntu-20.04
    services:
      postgres:
        env:
          POSTGRES_PASSWORD: postgres
        image: postgres:${{ matrix.postgres-version }}
        options: --health-cmd "pg_isready || psql -U postgres -c 'SELECT 1'"
        ports:
        - 5432:5432
    steps:
    - name: Checkout
      uses: actions/checkout@v2
      with:
        fetch-depth: 2
    - name: Install PHP
      uses: shivammathur/setup-php@v2
      with:
        coverage: pcov
        php-version: ${{ matrix.php-version }}
    - name: Cache dependencies installed with composer
      uses: actions/cache@v2
      with:
        key: php-${{ matrix.php-version }}-composer-locked-${{ hashFiles('composer.lock')
          }}
        path: ~/.composer/cache
        restore-keys: php-${{ matrix.php-version }}-composer-locked-
    - name: Install dependencies with composer
      run: composer install --no-interaction --no-progress --no-suggest
    - name: Run PHPUnit
      run: vendor/bin/phpunit -c ci/github/phpunit.pgsql.xml --coverage-clover=coverage.xml
    - name: Upload Code Coverage
      uses: codecov/codecov-action@v1
    strategy:
      matrix:
        php-version:
        - "7.4"
        postgres-version:
        - "9.2"
        - "9.3"
        - "9.4"
        - "9.5"
        - "9.6"
        - "10"
        - "11"
        - "12"
        - "13"
  phpunit-smoke-check:
    name: PHPUnit with SQLite
    runs-on: ubuntu-20.04
    steps:
    - name: Checkout
      uses: actions/checkout@v2
      with:
        fetch-depth: 2
    - name: Install PHP
      uses: shivammathur/setup-php@v2
      with:
        coverage: pcov
        php-version: ${{ matrix.php-version }}
    - name: Cache dependencies installed with composer
      uses: actions/cache@v2
      with:
        key: php-${{ matrix.php-version }}-composer-locked-${{ hashFiles('composer.lock')
          }}
        path: ~/.composer/cache
        restore-keys: php-${{ matrix.php-version }}-composer-locked-
    - if: ${{ matrix.deps == 'fixed' }}
      name: Install dependencies with composer
      run: composer install --no-interaction --no-progress --no-suggest
    - if: ${{ matrix.deps == 'low' }}
      name: Install lowest possible dependencies with composer
      run: composer update --no-interaction --no-progress --no-suggest --prefer-dist
        --prefer-lowest
    - name: Run PHPUnit
      run: vendor/bin/phpunit -c ci/github/phpunit.sqlite.xml --coverage-clover=coverage.xml
    - name: Upload Code Coverage
      uses: codecov/codecov-action@v1
    strategy:
      matrix:
        deps:
        - fixed
        include:
        - deps: low
          php-version: "7.3"
        php-version:
        - "7.3"
        - "7.4"
  static-analysis-phpstan:
    name: Static Analysis with PHPStan
    runs-on: ubuntu-20.04
    steps:
    - name: Checkout code
      uses: actions/checkout@v2
    - name: Install PHP
      uses: shivammathur/setup-php@v2
      with:
        coverage: none
        php-version: ${{ matrix.php-version }}
        tools: cs2pr
    - name: Cache dependencies installed with composer
      uses: actions/cache@v1
      with:
        key: php-${{ matrix.php-version }}-composer-locked-${{ hashFiles('composer.lock')
          }}
        path: ~/.composer/cache
        restore-keys: php-${{ matrix.php-version }}-composer-locked-
    - name: Install dependencies with composer
      run: composer install --no-interaction --no-progress --no-suggest
    - name: Run a static analysis with phpstan/phpstan
      run: vendor/bin/phpstan analyse --error-format=checkstyle | cs2pr
    strategy:
      matrix:
        php-version:
        - "7.4"
  static-analysis-psalm:
    name: Static Analysis with Psalm
    runs-on: ubuntu-20.04
    steps:
    - name: Checkout code
      uses: actions/checkout@v2
    - name: Install PHP
      uses: shivammathur/setup-php@v2
      with:
        coverage: none
        php-version: ${{ matrix.php-version }}
    - name: Cache dependencies installed with composer
      uses: actions/cache@v2
      with:
        key: php-${{ matrix.php-version }}-composer-locked-${{ hashFiles('composer.lock')
          }}
        path: ~/.composer/cache
        restore-keys: php-${{ matrix.php-version }}-composer-locked-
    - name: Install dependencies with composer
      run: composer install --no-interaction --no-progress --no-suggest
    - name: Run a static analysis with vimeo/psalm
      run: vendor/bin/psalm --show-info=false --stats --output-format=github --threads=4
    strategy:
      matrix:
        php-version:
        - "7.4"
```
</details>

Reply:
I think we elected to always use strings in other workflows, so let's stick with that, shall we?

Reply:
Yes, let's quote everything using double-quotes.